### PR TITLE
[LPT] Avoid using std::shared_ptr<Node> when creating a node

### PR DESCRIPTION
**Pull request by v-Golubev**
### Details:
 - *LP Transformations: fixed usage std::shared_ptr<Node> when creating a node (Except in cases with constant parent)*

### Tickets:
 - *63266*

### CODE REVIEW

**Comment by v-Golubev** [comment]
@eshoguli Could you please take a look?

File: inference-engine/src/low_precision_transformations/include/low_precision/depth_to_space.hpp [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
 #pragma once
 
-#include "transparent_base_transformation.hpp"
+#include "layer_transformation.hpp"
 
 namespace ngraph {
 namespace pass {
 namespace low_precision {
 
-class LP_TRANSFORMATIONS_API DepthToSpaceTransformation : public TransparentBaseTransformation {
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/include/low_precision/depth_to_space.hpp]: Done, thanks!

File: inference-engine/src/low_precision_transformations/src/pull_transpose_through_dequantization.cpp [hunk 2]
```diff
@@ -51,8 +51,8 @@ std::shared_ptr<Node> moveThroughElementwise(const std::shared_ptr<Node>& transp
         transposeValues }));
 
     const auto newElementwiseValues = ngraph::pass::low_precision::fold<opset1::Transpose>(
-        elementwiseValues->output(0),
-        transposeValues->output(0));
+        elementwiseValues,
+        transposeValues);
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/src/pull_transpose_through_dequantization.cpp:55]: great! thanks

**Review by eshoguli — APPROVED** [review]
LGTM, general comment: I'm little confused: in some cases you explicitly define output for `Constant` ops, in same cases you remove it. Let's not use `output(0)` explicitly if you have to pass default port for an operation: in other words - don't remove default port nGraph validation.

**Pull request merged**